src/pages/index.js

```diff
@@ -79,42 +148,133 @@ const addCardPopUp = new PopUpWithForm(cardPopUpSelector, handleAddCardSubmit);
 /** FUNCTIONS **/
 
 /* CARD */
-// Re - why we keep createCard in index.js again? Why not in Card.js?
 function createCard(cardData) {
-  const card = new Card(cardData, cardSelector, handleImageClick);
+  const card = new Card(
+    cardData,
+    cardSelector,
+    handleImageClick,
+    handleTrashClick,
+    handleLikeClick
+  );
   return card.generateCard();
 }
 
 /* EVENT HANDLERS */
 
 function handleProfileEditSubmit(data) {
-  user.setUserInfo(data.title, data.description);
+  api
+    .updateUserProfile({ name: data.title, about: data.description })
+    .then((res) => {
+      user.setUserInfo(res.name, res.about);
+      profileEditPopUp.close();
+    })
+    .catch((err) => {
+      console.error(`Profile Patch ${err}`);
+    })
+    .finally(() => {
+      profileEditPopUp.setSaving(false);
+    });
+
   profileEditPopUp.close();
 }
 
+function handleAvatarEditSubmit(data) {
+  api
+    .updateAvatar({ avatar: data.url })
+    .then((res) => {
+      user.setAvatar(res.avatar);
+      avatarEditPopUp.close();
+    })
+    .catch((err) => {
+      console.error(`Avatar Patch ${err}`);
+    })
+    .finally(() => {
+      avatarEditPopUp.setSaving(false);
+    });
+
+  avatarEditPopUp.close();
+  avatarFormValidator.disableButton();
+}
+
 function handleAddCardSubmit(cardInput) {
-  const cardEl = createCard({ name: cardInput.title, link: cardInput.url });
-  cardSection.addItem(cardEl);
+  //card.js has name and link, vs index.html has title and url
+  api
+    .addCard({ name: cardInput.title, link: cardInput.url })
+    .then((res) => {
+      const cardEl = createCard(res);
+      cardSection.addItem(cardEl);
+    })
+    .catch((err) => {
+      console.error(`Add Card ${err}`);
+    })
+    .finally(() => {
+      addCardPopUp.setSaving(false);
+    });
+
   addCardPopUp.close();
   addFormValidator.disableButton();
```

Comment:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.